.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,29 @@
+name: pre-commit
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  pre-commit:
+
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pre-commit
+          pip install -r requirements.txt
+
+      - name: Install pre-commit hooks
+        run: |
+          pre-commit install
```

Comment:
@mmanzoorTT tt-torch does not have these two steps, chat-gpt recommended them to me. Do you think we should have them in both repos?

Reply:
@uazizTT has implemented these workflows. He can provide more insight about it. Moreover, `pre-commit` package is part of `requirements.txt`; so adding `pip install pre-commit` seems redundant to me.

Reply:
@kmitrovicTT I agree with @mmanzoorTT that it is redundant to add these as they are part of requirements. Also for tt-torch, I am working on a change that will pre-build all dependencies in a single build machine and then copy it over to all the test runners so that all test runners don't have to build the environment again. For that reason I think it is better to put all dependencies in a single requirements.txt file.